examples/randlanet_classification.py

```diff
@@ -0,0 +1,255 @@
+
+    def forward(self, x, pos, batch):
+        edge_index = knn_graph(pos, self.num_neighbors, batch=batch, loop=True)
+
+        shortcut_of_x = self.shortcut(x)  # N, d_out
+        x = self.mlp1(x)  # N, d_out//8
+        x = self.lfa1(edge_index, x, pos)  # N, d_out//2
+        x = self.lfa2(edge_index, x, pos)  # N, d_out//2
+        x = self.mlp2(x)  # N, d_out
+        x = self.lrelu(x + shortcut_of_x)  # N, d_out
+
+        return x, pos, batch
+
+
+def get_decimation_idx(ptr, decimation):
+    """Subsamples each point cloud by a decimation factor.
+
+    Decimation happens separately for each cloud to prevent
+    emptying smaller point clouds.
+
+    """
+    batch_size = ptr.size(0) - 1
+    num_nodes = torch.Tensor([ptr[i + 1] - ptr[i]
+                              for i in range(batch_size)]).long()
+    decimated_num_nodes = num_nodes // decimation
+    idx_decim = torch.cat(
+        [(ptr[i] + torch.randperm(decimated_num_nodes[i], device=ptr.device))
+         for i in range(batch_size)],
+        dim=0,
+    )
+    # Update the ptr for future decimations
+    ptr_decim = ptr.clone()
+    for i in range(batch_size):
+        ptr_decim[i + 1] = ptr_decim[i] + decimated_num_nodes[i]
+    return idx_decim, ptr_decim
+
+
+def decimate(b_out, ptr, decimation):
+    decimation_idx, decimation_ptr = get_decimation_idx(ptr, decimation)
+    b_out_decim = tuple(t[decimation_idx] for t in b_out)
+    return b_out_decim, decimation_ptr
+
+
+class Net(torch.nn.Module):
+    def __init__(
+        self,
+        num_features,
+        num_classes,
+        decimation: int = 4,
+        num_neighboors: int = 16,
+        return_logits: bool = False,
+    ):
+        super().__init__()
+        self.decimation = decimation
+        self.return_logits = return_logits
+        self.fc0 = Sequential(Linear(in_features=num_features, out_features=8))
+        # 2 DilatedResidualBlock converges better than 4 on ModelNet.
+        self.block1 = DilatedResidualBlock(num_neighboors, 8, 32)
+        self.block2 = DilatedResidualBlock(num_neighboors, 32, 128)
+        self.mlp1 = SharedMLP([128, 128])
+        self.pool = GlobalPooling()
+        self.mlp_end = Sequential(SharedMLP([128, 32], dropout=[0.5]),
+                                  Linear(32, num_classes))
+
+    def forward(self, batch):
+        ptr = batch.ptr.clone()
+
+        in_0 = (self.fc0(batch.x), batch.pos, batch.batch)
+
+        b1 = self.block1(*in_0)
+        b1_decimated, ptr = decimate(b1, ptr, self.decimation)
+
+        b2 = self.block2(*b1_decimated)
+        b2_decimated, _ = decimate(b2, ptr, self.decimation)
+
+        x = self.mlp1(b2_decimated[0])
+        x, _, _ = self.pool(x, b2_decimated[1], b2_decimated[2])
+
+        logits = self.mlp_end(x)
+        if self.return_logits:
+            return logits
+        probas = logits.log_softmax(dim=-1)
+        return probas
+
+
+class SetPosAsXIfNoX(BaseTransform):
```

Comment:
Let's move this to the model forward pass?
```python
x = x if x is not None else pos
```

Reply:
done.